workers/unity/Packages/com.improbable.gdk.querybasedinteresthelper/Constraint.cs

```diff
@@ -0,0 +1,277 @@
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.Core;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+namespace Improbable.Gdk.QueryBasedInterest
+{
+    /// <summary>
+    ///     Utility class to help construct QueryConstraint objects.
+    /// </summary>
+    public static class Constraint
+    {
+        private static readonly List<ComponentInterest.QueryConstraint> EmptyList
```

Comment:
The fact that this isn't a property scares me slightly. As its now setting a reference to this static list when its used.

If someone manually changes the list inside a `ComponentInterest.*Constraint` that was set to `EmptyList` it will mutate this list.

Reply:
hmm yeah that's not very nice, but the trade-off then is to create 2 new empty lists on each new Constraint made